### PR TITLE
feat: track http request data in http trigger

### LINKED ISSUE #443: Add request telemetry tracking for out-of-process Azure Functions' `HttpRequestData`
**Is your feature request related to a problem? Please describe.**
When tracking HTTP requests in HTTP triggered Azure Functions that run out-of-process, there is no `HttpRequest` available, only an `HttpRequestData` instance.

**Describe the solution you'd like**
Add a dedicated Azure Functions package that lets use track HTTP requests with Azure Functions' `HttpRequestData`.

### PR DESCRIPTION
**Pull request by stijnmoreels**
Adds HTTP request tracking for isolated Azure Functions HTTP trigger's `HttpRequestData`.

Closes #443

### CODE REVIEW

File: docs/preview/03-Features/writing-different-telemetry-types.md [hunk 1]
```diff
@@ -593,6 +593,50 @@ using (var measurement = DurationMeasurement.Start())
 
 > 💡 Note that [Arcus Web API request tracking middleware](https://webapi.arcus-azure.net/features/logging#logging-incoming-requests) can already do this for you in a ASP.NET Core application
 
+### Incoming HTTP requests in Azure Function HTTP trigger
```
Comment (fgheysels) [inline docs/preview/03-Features/writing-different-telemetry-types.md:596]:
```suggestion
### Incoming HTTP requests in HTTP triggered Azure Functions
```

File: docs/preview/03-Features/writing-different-telemetry-types.md [hunk 2]
```diff
@@ -593,6 +593,50 @@ using (var measurement = DurationMeasurement.Start())
...
+Requests allow you to keep track of the HTTP requests that are performed against your Azure Function HTTP trigger and what the response was that was sent out.
+
+**Installation**
+
+If you want to track the `HttpRequestData` of an isolated Azure Functions HTTP trigger project, you'll have to install an additional package to include these dependencies:
```
Comment (fgheysels) [inline docs/preview/03-Features/writing-different-telemetry-types.md:601]:
```suggestion
For isolated Azure Functions, you'll have to install an additional package if you want to track the `HttpRequestData`:
```

**Review by fgheysels — APPROVED** [review]

File: docs/preview/03-Features/writing-different-telemetry-types.md (hunk 1 above)
Comment (stijnmoreels) [reply docs/preview/03-Features/writing-different-telemetry-types.md:596]: Is that better? 'triggered functions' doesn't seem like an improvement here?

Comment (fgheysels) [reply docs/preview/03-Features/writing-different-telemetry-types.md:596]: Well, I was thinking:  the Azure Function is receiving the http request (it is triggered by an http trigger).  It is not the HTTP Trigger that receives the request.

Comment (stijnmoreels) [reply docs/preview/03-Features/writing-different-telemetry-types.md:596]: It's maybe more to read like Service Bus topic, for example.
Microsoft uses this in their title: https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-http-webhook-trigger?tabs=in-process%2Cfunctionsv2&pivots=programming-language-csharp

Comment (fgheysels) [reply docs/preview/03-Features/writing-different-telemetry-types.md:596]: That's really documentation on the Http Trigger (binding) of a Function.  So, imho, that is something different.
But, for me it is OK.  I just thought it was clearer / more correct.

**Comment by fgheysels** [comment]
@stijnmoreels , there are some conflicts.  Can you have a look ?

File: docs/preview/03-Features/writing-different-telemetry-types.md (hunk 1 above)
Comment (stijnmoreels) [reply docs/preview/03-Features/writing-different-telemetry-types.md:596]: When we complete our service correlation docs we could maybe agree on a common naming convention we use for pur components and update the docs in the different libraries accordingly. 👍

**Pull request merged**